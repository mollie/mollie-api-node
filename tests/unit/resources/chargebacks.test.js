import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

import Chargebacks from 'resources/chargebacks';

import response from '../__stubs__/chargebacks.json';

const mock = new MockAdapter(axios);

describe('chargebacks', () => {
  let chargebacks;

  beforeEach(() => {
    chargebacks = new Chargebacks(axios.create());
  });

  describe('.all()', () => {
    mock.onGet('/chargebacks').reply(200, response);

    it('should return a list of all chargebacks', () =>
      chargebacks.all().then(result => {
        expect(result).toBeInstanceOf(Array);
        expect(result).toHaveProperty('links');
        expect(result).toMatchSnapshot();
      }));

    it('should work with a callback', done => {
      chargebacks.all((err, result) => {
        expect(err).toBeNull();
        expect(result).toBeInstanceOf(Array);
        expect(result).toHaveProperty('links');
        expect(result).toMatchSnapshot();
        done();
      });
    });
  });
});
