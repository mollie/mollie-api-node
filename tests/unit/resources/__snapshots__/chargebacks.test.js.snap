// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chargebacks .all() should return a list of all chargebacks 1`] = `
Array [
  Chargeback {
    "_links": Object {
      "documentation": Object {
        "href": "https://docs.mollie.com/reference/v2/chargebacks-api/get-chargeback",
        "type": "text/html",
      },
      "payment": Object {
        "href": "https://api.mollie.com/v2/payments/tr_WDqYK6vllg",
        "type": "application/hal+json",
      },
      "self": Object {
        "href": "https://api.mollie.com/v2/payments/tr_WDqYK6vllg/chargebacks/chb_n9z0tp",
        "type": "application/hal+json",
      },
    },
    "amount": Object {
      "currency": "USD",
      "value": "43.38",
    },
    "createdAt": "2018-03-14T17:00:52.0Z",
    "id": "chb_n9z0tp",
    "paymentId": "tr_WDqYK6vllg",
    "resource": "chargeback",
    "reversedAt": null,
    "settlementAmount": Object {
      "currency": "EUR",
      "value": "35.07",
    },
  },
]
`;

exports[`chargebacks .all() should work with a callback 1`] = `
Array [
  Chargeback {
    "_links": Object {
      "documentation": Object {
        "href": "https://docs.mollie.com/reference/v2/chargebacks-api/get-chargeback",
        "type": "text/html",
      },
      "payment": Object {
        "href": "https://api.mollie.com/v2/payments/tr_WDqYK6vllg",
        "type": "application/hal+json",
      },
      "self": Object {
        "href": "https://api.mollie.com/v2/payments/tr_WDqYK6vllg/chargebacks/chb_n9z0tp",
        "type": "application/hal+json",
      },
    },
    "amount": Object {
      "currency": "USD",
      "value": "43.38",
    },
    "createdAt": "2018-03-14T17:00:52.0Z",
    "id": "chb_n9z0tp",
    "paymentId": "tr_WDqYK6vllg",
    "resource": "chargeback",
    "reversedAt": null,
    "settlementAmount": Object {
      "currency": "EUR",
      "value": "35.07",
    },
  },
]
`;
